avail/src/main/kotlin/avail/resolver/FileSystemModuleRootResolver.kt

```diff
@@ -498,14 +465,18 @@ class FileSystemModuleRootResolver constructor(
 		withContents: (ByteArray, UUID?)->Unit,
 		failureHandler: (ErrorCode, Throwable?)->Unit)
 	{
-		if (setOf(ROOT, DIRECTORY, PACKAGE).contains(reference.type))
+		when(reference.type)
 		{
-			failureHandler(
-				StandardErrorCode.IO_EXCEPTION,
-				IOException(
-					"${reference.qualifiedName} is a directory, not a file",
-					null))
-			return
+			Root, Directory, is Package ->
```

Comment:
Interesting, I don't think I've seen that notation before for mixing singletons and subtypes.

Reply:
Upon further examination, it looks like using `.isDirectory` might be more maintenance-friendly here.